src/js/polygon.js

```diff
@@ -0,0 +1,149 @@
+import * as MathService from "./math.service";
+
+class Polygon {
+    constructor(firstX, firstY, width, height) {
+        this._firstVertex = {
+            x: firstX,
+            y: firstY
+        };
+        this._vertices = [];
+        this._width = width;
+        this._height = height;
+        this._isIntersected = false;
+        this._intersections = [];
+        this._isSnapped = false;
+
+        this.__initialize();
+    }
+
+    get vertices() {
+        return this._vertices;
+    }
+
+    get firstVertex() {
+        return this._firstVertex;
+    }
+
+    get centerPoint() {
+        return this._centerPoint;
+    }
+    
+    get isIntersected() {
+        return this._isIntersected;
+    }
+
+    get isSnapped() {
+        return this._isSnapped;
+    }
+
+    get intersections() {
+        return this._intersections;
+    }
+
+    set isIntersected(value) {
+        this._isIntersected = value;
+    }
+
+    set isSnapped(value) {
+        this._isSnapped = value;
+    }
+
+    __initialize() {
```

Comment:
how about create nested Class `Polygon` and move init into it and some methods like `getTopLine`